lib/Runtime/Library/JSON.cpp

```diff
@@ -688,55 +689,45 @@ namespace JSON
 
                         result = Js::ConcatStringBuilder::New(this->scriptContext, propertyCount);
                         Js::DynamicTypeHandler * typeHandler = dynamicObject->GetTypeHandler();
-                        enumerator.Reset();
-                        Js::PropertyId previousId;
+
                         // if object has an objectarray, (aka indexed properties)
                         // we need to loop them first since they won't show up on the second loop below
                         if (dynamicObject->HasObjectArray())
                         {
-                            int totalNumPropertyCount = this->GetPropertyCount(object, &enumerator) - propertyCount;
-                            Assert(totalNumPropertyCount >= 0);
-
-                            int index = 0;
                             enumerator.Reset();
-                            while (index < totalNumPropertyCount && (propertyName = enumerator.MoveAndGetNext(id)) != NULL)
+                            while ((propertyName = enumerator.MoveAndGetNext(id)) != NULL)
                             {
                                 // if unsuccessful get propertyId from the string
-                                scriptContext->GetOrAddPropertyRecord(propertyName->GetString(), propertyName->GetLength(), &propRecord);
+                                scriptContext->GetOrAddPropertyRecord(propertyName, &propRecord);
+                                if (!propRecord->IsNumeric()) break;
                                 id = propRecord->GetPropertyId();
                                 StringifyMemberObject(propertyName, id, value, (Js::ConcatStringBuilder*)result, indentString, memberSeparator, isFirstMember, isEmpty);
-                                index++;
                             }
                         }
 
                         // loop over the named properties
                         for (uint32 i = 0; i < propertyCount; i++)
                         {
                             id = typeHandler->GetPropertyId(scriptContext, (Js::PropertyId)i);
-
-                            if (id < Js::InternalPropertyIds::Count)
+                            if (id == Js::Constants::NoProperty)
                             {
-                                // if the property ID is internal, we cannot assume dynamicObject->getSlot(i) is a var later
-                                previousId = id;
                                 continue;
```

Comment:
Why is this case now a no-op? Previously (both immediately prior to this PR and also before your original stringify changes) this case tried to fetch the id from the enumerator

Reply:
because, now implicitly checks `enumerable` and `callable` on premises while reading the property using typeHandler lookup.